### ISSUE TITLE
Pet Colors

### ISSUE THREAD

**Issue by PThagaard** [DSH105/EchoPet#478]
Black Wolf, changes to white wolf.

Red to blue and so on.

The color changes - but the selected color and the showed color - are 2 very different things.

**Comment by PThagaard** [comment]
Wolf's:

Black = White
Blue=Yellow
Brown=Light Blue
Cyan=Pink
Gray=Light Gray
Green=Magenta
Lime=Purple
Orange=Red

**Comment by DSH105** [comment]
Not sure I understand what you mean here.

**Comment by South-Paw** [comment]
I've just spent a bit of time trying to figure out where the problem is (in hope of fixing it for you) but I can't find what to blame for it....

What this issue relates to though is the Wolfs collar color menu; the colored wool you click on doesn't give the correct colored collar.

Eg: clicking on black wool for a collar gives a white collar and clicking on the white wool for a white collar gives a black collar.

**Comment by South-Paw** [comment]
@DSH105 My comment above hopefully identifies the issue a bit better

**Comment by DSH105** [comment]
Yeah, thank you :). I haven't really looked into it that much yet though.

**Comment by JamieSinn** [comment]
More information @DSH105 http://www.ecocitycraft.com/forum/threads/pet-wolf-collar-colour-bug.129504/

**Comment by DSH105** [comment]
I believe @Harry5573 may have resolved this in #539 (https://github.com/DSH105/EchoPet/pull/539/files#diff-89ba8f1e527bd5e10ad0d0d4563647adR88). I haven't drafted a new minor release yet, but there are development builds available if anyone is willing to test it.

**Comment by Harry5573** [comment]
@DSH105 I fixed it v1_7_R1-v1_7_R4 for 1.7, and also added a converter in the protocol-hack listener to fix it for 1.8 clients on v1_7_R4 (https://github.com/Harry5573/EchoPet/blob/ca601dbee5d32e206500ad17acd8862bb6914d43/modules/EchoPet/src/main/java/com/dsh105/echopet/compatibility/SpigotProtocolHackPacketListener.java#L143), i didin't touch 1.8 spigot build modules though, so could still be broken there.

**Comment by DSH105** [comment]
Above commit should bring the `1.8.x` modules back into line with NMS.

**Comment by klugemonkey** [comment]
Collars still not working 1.8.3+
modules\EchoPet\pom.xml is missing dependency for EchoPet-v1_8_R3 and
Dye enums (I believe used for collars) are reverse of Wool enums ->  (byte) 15 - dc.getWoolData()

**Comment by klugemonkey** [comment]
This fixes the issue on 1.8.3+:

    @Override
    public void setCollarColor(DyeColor dc) {
        if (((IWolfPet) pet).isTamed()) {
        	int colour = 15 - (int)dc.getWoolData();
            this.datawatcher.watch(20, Byte.valueOf((byte)colour));
        }
    }
